hudi-common/src/main/java/org/apache/hudi/common/config/HoodieMetadataConfig.java

```diff
@@ -137,6 +136,12 @@
       .withDocumentation("There are cases when extra files are requested to be deleted from metadata table which was never added before. This config"
           + "determines how to handle such spurious deletes");
 
+  public static final ConfigProperty<Boolean> RECORDKEY_DE_DUPLICATE = ConfigProperty
+      .key(METADATA_PREFIX + ".recordkey.deduplicate")
+      .defaultValue(true)
+      .sinceVersion("0.10.0")
+      .withDocumentation("When enabled, Metadata table records will have key de-duplication.");
```

Comment:
can we make this doc clearer? IMO this has to be an internal config. i.e `"_." + METADATA_PREFIX + ".recordkey.deduplicate"` that the user should not normally bother with

Reply:
added the internal config prefix.